### PR TITLE
Create hook to subscribe to hash 

### PR DESCRIPTION
**Pull request by ghana7989**

### CODE REVIEW

File: src/mantine-hooks/src/use-id/use-id.test.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ describe('@mantine/hooks/use-id', () => {
   });
 
   it('returns generated id if id was not provided', () => {
-    const hook = renderHook(() => useId(null, () => 'test-id'));
+    const hook = renderHook(() => useId(undefined, () => 'test-id'));
```
Comment (rtivital) [inline src/mantine-hooks/src/use-id/use-id.test.ts:16]: Please revert this change and do not change files apart from your pr scope

**Comment by rtivital** [comment]
Hi, thanks for pull request, make sure to complete all other todo items in trello board so that this can be merged – https://trello.com/c/HmCPGNgS/20-mantine-hooks-use-hash

File: src/mantine-hooks/src/use-id/use-id.test.ts (hunk 1 above)
Comment (ghana7989) [reply src/mantine-hooks/src/use-id/use-id.test.ts]: Okay but there was a typescript error.
changing it to undefined fixed it

**Comment by ghana7989** [comment]
Can You assign someone for writing the docs?
Because I am not sure about it.

**Comment by rtivital** [comment]
It's not that hard, please create basic docs:
- demo for hook (button when clicked sets hash to random string, see use-document-title hook)
- docs page – use-hash.mdx with meta information and demo

You can leave descriptions blank, if you are struggling with describing it in English – that's fine

**Comment by ghana7989** [comment]
> It's not that hard, please create basic docs:
> - demo for hook (button when clicked sets hash to random string, see use-document-title hook)
> - docs page – use-hash.mdx with meta information and demo
> 
> You can leave descriptions blank, if you are struggling with describing it in English – that's fine

Okay then I will try my best 😊.

**Comment by rtivital** [comment]
@ghana7989 Please ping me here once you are ready to submit pr for code review

**Comment by ghana7989** [comment]
It is ready for review but I didn't add tests as I don't know how to add them and also it is not mentioned in trello board

**Comment by rtivital** [comment]
@ghana7989 can you also add demo? Like in use-document-title hook

**Comment by ghana7989** [comment]
Can you tell me the folders.
Because I already add docs by creating use-hash.mdx

**Comment by rtivital** [comment]
Hooks demos [are here](https://github.com/mantinedev/mantine/tree/master/docs/src/demos), see usage example with use-document-title hook

**Comment by ghana7989** [comment]
I think this is for Final Review
I have updated docs and also created use-hash.tsx file and I tested, this is completely working as expected.

**Comment by rtivital** [comment]
Alright looks good, last thing to do is to resolve conflict with next minor branch and it can be merged

**Comment by rtivital** [comment]
Looks good, thanks!
